cdci_data_analysis/analysis/queries.py

```diff
@@ -316,6 +316,8 @@ def __init__(self,
 
         super(InstrumentQuery, self).__init__(name,parameters_list)
 
+    def check_instrument_roles(self, roles):
+        raise RuntimeError(f'{self}: check_instrument_access needs to be implemented for the {self.name} instrument')
```

Comment:
this should check for development role in the `roles`, for the `development` attribute in the instance, and return accordingly. This is the useful default, no?

Reply:
actually only checking the development role is within `roles`

development attribute is checked outside

Reply:
if error raising will stay the default here, error message is misleading: `check_instrument_access` actually is implemented and it checks for the attribute. One must implement this actual method

Reply:
Default should be compatible with what we had before, i.e.:

```python
(self.development and ('oda workflow developer' in roles)) or (not self.development)
```

By the way, maybe good to already now pass also the tocket subject (email) to the method. We will use it in the future.

Reply:
> if error raising will stay the default here, error message is misleading: `check_instrument_access` actually is implemented and it checks for the attribute. One must implement this actual method

you're right I forgot to change the message once renamed the method